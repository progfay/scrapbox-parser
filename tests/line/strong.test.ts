describe('strong', () => {
  it('Simple strong', () => {
    expect('[[Simple strong]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('[[]] is not strong', () => {
    expect('[[]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Decoration in Strong notation', () => {
    expect('[[[! deco]]]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })
})
