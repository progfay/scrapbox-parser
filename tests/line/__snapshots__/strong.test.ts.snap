// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strong [[]] is not strong: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "[[]]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strong Decoration in Strong notation: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "[[",
        "type": "plain",
      },
      Object {
        "decos": Array [
          "!",
        ],
        "nodes": Array [
          Object {
            "text": "deco",
            "type": "plain",
          },
        ],
        "type": "decoration",
      },
      Object {
        "text": "]]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strong Simple strong: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "nodes": Array [
          Object {
            "text": "Simple strong",
            "type": "plain",
          },
        ],
        "type": "strong",
      },
    ],
    "type": "line",
  },
]
`;
